### DISCUSSION TITLE
ref class

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/csharplang#7097] [General]
```ref class``` is basically consistent with ```ref struct```, except that ```ref class``` is class and ```ref struct``` is struct.

For example
```
public ref class RC { }
```
```
public class C { }
public ref struct RS<T> where T : class { public T t; }
```

```RC``` is equivalent to ```RS<C>```. The reference of ```RC```, which is the same as ```RS<C>```, can only exist in one thread stack, so I don't have to consider the issue of multithreading concurrency for ```RC```.

**Comment by HaloFour** [comment]
A `ref struct` is a `struct` that can only live on the stack.  As `class`es always live on the heap and not the stack I don't see a point in a `ref class`.

**Reply by ghost** [reply]
But the reference of ```ref class``` always exist in one thread stack.

**Reply by HaloFour** [reply]
`ref struct` doesn't limit how many references to a given struct are taken.  I understand what you're trying to accomplish, in that you only want a single reference to a given class instance, however that's not equivalent to what `ref struct` is intended to accomplish.
